ethgas/ethgas.go

```diff
@@ -136,8 +136,19 @@ func (g *GasGauge) run() error {
 			ema70.Tick(new(big.Int).SetUint64(p70))
 			ema95.Tick(new(big.Int).SetUint64(p95))
 
-			standard := ema30.Value().Uint64()
-			slow := float64(standard) * 0.85
+			// calculate the suggested slow price; taking unused gas into account
+			var slow float64
+			gasUnused := latestBlock.GasLimit() - latestBlock.GasUsed()
+			avgTxSize := latestBlock.GasUsed() / uint64(len(txns))
+			if gasUnused >= avgTxSize {
+				// The block had space left over for more transactions, meaning
+				// that a transaction with a very low gas price could have been
+				// mined
+				slow = 1e9
```

Comment:
Philippe suggested using the network's minimum gas price, but I couldn't find that in the `go-ethereum` types so I hard-coded it instead

Reply:
I don't think setting slow to a constant value isn't the ideal algorithm here. All Fast/Standard/Slow should move with a EMA as designed, nothing is constant/point-in-time, everything is a moving.

Another idea is to calculate the % of unused gas from the total.. ie, `latestBlock.GasUsed() / latestBlock.GasLimit()`, and then use this as a multiplier to lower the price, ie. whatever fast/standard/slow calculations are, in the very end, multiply by that number. In which case, if full blocks then will be multiplied by 1, if half blocks, then we're suggesting half the price. The idea as well in this algorithm is that we're contributing "weights" into a value that is calculated / smoothed over time.